src/lib/adapters/waku/waku.ts

```diff
@@ -57,7 +61,10 @@ export async function connectWaku(options?: ConnectWakuOptions) {
 	})
 
 	await waku.start()
-	await waku.dial(peerMultiaddr)
+	for (const peer of peers) {
+		const addr = multiaddr(peer)
+		await waku.dial(addr)
+	}
```

Comment:
What about something like:
```suggestion
	await Promise.race(peers.map(peer => waku.dial(multiaddr(peer)))
```
This way it will continue execution after one peer is connected, but still try to connect to the other ones in the background.

Reply:
I thought of that but then there is also an async call to `waitForRemotePeer` after that. I am not sure how the combination of these things work and need to further look into their source code.

Another problem is that this may cause an unhandled rejection if the `dial` of a non-winning peer fails for some reason. I am trying to be more mindful of these issues after experiencing similar bugs from js-waku.

Reply:
https://github.com/logos-innovation-lab/waku-objects-playground/issues/404